### DISCUSSION TITLE
I am getting a difference in quality compared to A1111. Is there a way to correct this?

### DISCUSSION THREAD

**Discussion by Javicshz** [vladmandic/sdnext#334] [General]

**Comment by vladmandic** [comment]
can you upload actual image generated in a1111 and using this repo so i can take a look at their metadata?

**Reply by vladmandic** [reply]
sorry for a dumb question, but what is `easynegative`?

**Reply by vladmandic** [reply]
that answers it - download `easynegative` and place it embeddings folder, it will be recognized and used.
but note that default embeddings folder here is NOT `/embeddings`, its `/models/embeddings` (and you can change it via UI -> System Paths)

so i'm guessing that answers this issue?

**Comment by qpl23** [comment]
I saw the [reddit thread](https://www.reddit.com/r/StableDiffusion/comments/12tuf2y/the_quality_between_a1111_and_vlads_fork_are/) on this and did the experiment locally. My result was an image indistinguishable from OP's 1111 image (to my eyes): 
![swede1](https://user-images.githubusercontent.com/47281117/233648322-1454deb4-faf2-481b-a5a8-d8afa8dc3aae.png)

The image looks so similar I thought I should generate a similar, related image to show I didn't just copy OP's image, so here's one with one extra generation step which has morphed the background telephone pole into a tree, for example:
![swede2](https://user-images.githubusercontent.com/47281117/233650091-93f18303-455f-44b7-91d5-fcbd0b998421.png)

These were generated with OP's model / prompt / params from the reddit thread linked previously. Metadata intact at time of uploading.

(Edit to add: in case it makes any difference, my install is a couple of days out of date; git log shows last commit as  "Tue Apr 18 08:06:16 2023 -0400 / disable gradio queues on demand")

**Comment by myndxero** [comment]
There are a lot of reasons things can vary. I noticed that Vlad's has some of the k sampler stuff checked by default that auto1111's does not (I think, iirc). Check clip skip and noise delta too. Some screenshots of various things that could be different that would result in a different generation even with otherwise identical settings.

![image](https://user-images.githubusercontent.com/113237671/233658320-5d78b31f-63e2-4374-bfd2-303de2bb7603.png)
![image](https://user-images.githubusercontent.com/113237671/233658443-b47309a5-56b3-4e75-b570-f42806898f04.png)
![image](https://user-images.githubusercontent.com/113237671/233660899-9a38c1b8-5144-43fe-84b4-266016f350a9.png)
![image](https://user-images.githubusercontent.com/113237671/233660994-6a81fd27-66c1-405d-a2d3-1f3df20d57ff.png)

**Reply by Nrgte** [reply]
I'm also getting vastly different images with vladmandic than in a1111. 
I've checked all the settings in both UIs and they seem identical. I have --no-half in a1111 so I enabled that in the CUDA settings for vlads. 

I've checked the settings in Stable Diffusion, CUDA Settings, Sampler Parameters and I use the same VAE, model, prompts, seed.

I ddin't find a Compatibilty tab equivalent in vlad.

Does anyone know how to debug this further?

**Reply by Javicshz** [reply]
Vlads uses different locations for embeddings. Make sure you place embedding in /automatic/models/embeddings and not automatic/embeddings. This is what fixed it for me.

Same goes for loras. They should be placed in \automatic\extensions\sd-webui-additional-networks\models\lora

**Reply by Nrgte** [reply]
I don't use any embeddings, but I have the loras in models\Lora instead of extensions\sd-webui-additional-networks\models\lora
But adding them there as well didn't make any difference.

**Comment by myndxero** [comment]
I don't remember which ones, but the compatibility tab boxes were merged
into another section.

Do you discard the next to last sigma or use a different noise delta? Vae
on auto or overriding built in Vaes anything like that?

**Reply by Nrgte** [reply]
No I don't discard the next to last sigma. But that was also the case in a1111:
![grafik](https://user-images.githubusercontent.com/131605234/233856751-14b35330-9ecb-426a-94ae-52bae3bc4b7b.png)

Noise delta is on 0 on both as well.

![grafik](https://user-images.githubusercontent.com/131605234/233856836-c313d072-d77d-4840-8f56-88d750247749.png)

I'm unsure about the CUDA settings, since those didn't exist. And I can't find the Compatibility settings. Otherwise everything seems the same.

**Comment by myndxero** [comment]
They're mixed in, compat settings, in Vlad. Idr what they were or I could
help find.

**Reply by vladmandic** [reply]
most cuda settings in a1111 are specified via command line params (for example `--no-half` is cmd flag in a1111 and its a setting in ui here)
